File: C/SelectionSort.c
// BubbleSort.c
#include <stdio.h>

void swap(int* a, int* b) {
    int t = *a;
    *a = *b;
    *b = t;
}

void selectionSort(int arr[], int n) {
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++) {
        min_idx = i;
        for (j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        swap(&arr[min_idx], &arr[i]);
    }
}

int main() {
    int n;
    // Lê o número de elementos
    scanf("%d", &n);
    int arr[n];
    // Lê os elementos
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // printf("Array original:\n");
    // printArray(arr, n);
    
    selectionSort(arr, n);
    
    // printf("Array ordenado (Bubble Sort):\n");
    // printArray(arr, n);
    return 0;
}

File: Java/SelectionSort.java
import java.util.Scanner;

// BubbleSort.java
public class SelectionSort {
        public static void selectionSort(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[minIndex]) {
                    minIndex = j;
                }
            }
            int temp = array[minIndex];
            array[minIndex] = array[i];
            array[i] = temp;
        }
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Lê o número de elementos
        int n = scanner.nextInt();
        int[] arr = new int[n];
        // Lê os elementos
        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        scanner.close();

        // System.out.println("Array original:");
        // printArray(arr);
        selectionSort(arr);
        // System.out.println("Array ordenado (Bubble Sort):");
        // printArray(arr);
    }
}
